### ISSUE TITLE
The different outputs using metaspades.py with the same k-mer

### ISSUE THREAD

**Issue by SarahIsme** [ablab/spades#275]
[67_77_87.pdf](https://github.com/ablab/spades/files/3046307/67_77_87.pdf)
[87_93_95.pdf](https://github.com/ablab/spades/files/3046308/87_93_95.pdf)
Dear developers.

Please check the attached files. I used metaspades.py with k-mer as 67,77,87 to get the contigs.fasta of 87, but the statistical characteristics of this file are different from the contigs.fasta of 87 in another run, where k-mer was set as 87,93,95.

Which should be more accurate?

Another question, when I confirm the best k-mer, could I assemble the reads using the only k-mer value not multi-kmers?

Thanks in advance.

Best regards,

Sara

**Comment by asl** [comment]
Dear @SarahIsme 

These are not the contigs you're looking for. Per SPAdes manual (http://cab.spbu.ru/files/release3.13.0/manual.html#sec3.5):
```
SPAdes stores all output files in <output_dir> , which is set by the user.

<output_dir>/corrected/ directory contains reads corrected by BayesHammer in *.fastq.gz files; if compression is disabled, reads are stored in uncompressed *.fastq files
<output_dir>/scaffolds.fasta contains resulting scaffolds (recommended for use as resulting sequences)
<output_dir>/contigs.fasta contains resulting contigs
<output_dir>/assembly_graph.gfa contains SPAdes assembly graph and scaffolds paths in GFA 1.0 format
<output_dir>/assembly_graph.fastg contains SPAdes assembly graph in FASTG format
<output_dir>/contigs.paths contains paths in the assembly graph corresponding to contigs.fasta (see details below)
<output_dir>/scaffolds.paths contains paths in the assembly graph corresponding to scaffolds.fasta (see details below)
```

and later on:
```
    K<##>/ – directory containing intermediate files from the run with K=<##>. These files should not be used as assembly results; use resulting contigs/scaffolds in files mentioned above.
```

> Which should be more accurate?

You cannot judge this simply from lengths statistics. You need a reference genome.

> Another question, when I confirm the best k-mer, could I assemble the reads using the only k-mer value not multi-kmers?

There is no *best k-mer*. SPAdes *is* a multi-k-mer assembler and therefore the whole set of k-mers should be used. Also, it's not wise to skip the short k-mers especially for low-covered metagenome assemblies.

Long story short: do not change k-mer length settings unless you're 100% sure what you're doing. Otherwise you're likely obtain some intermediate files.
